### PR TITLE
Fix crash in lua host lookup

### PR DESCRIPTION
**Pull request by shukitchan**
I think I miss reading the documentation - https://docs.trafficserver.apache.org/en/latest/developer-guide/api/functions/TSHostLookup.en.html

I was not aware of the case when the call can be synchronous

Also we need to yield the luaVM when we need to wait for an answer. And we should not yield if the call is synchronous. And then in the handler, we need to see if we have yielded or not. So we can resume only if we have yielded.

### CODE REVIEW

**Review by bneradt — COMMENTED** [review]
Makes sense to me.

File: plugins/lua/ts_lua_misc.c [hunk 1]
```diff
@@ -444,30 +444,42 @@ ts_lua_host_lookup(lua_State *L)
...
   lua_State *L;
   ts_lua_coroutine *crt;
+  unsigned int resume;
 
   ai  = TSContDataGet(contp);
   ci  = ai->cinfo;
   crt = &ci->routine;
   L   = crt->lua;
 
+  // find out if need to resumt luaVM before async item cleanup
```
Comment (randall) [inline plugins/lua/ts_lua_misc.c:475]: typo (resumt)

Comment (shukitchan) [reply plugins/lua/ts_lua_misc.c]: fixed

**Review by randall — APPROVED** [review]

**Comment by zwoop** [comment]
Cherry-picked to v9.1.x branch.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Comment by ezelkow1** [comment]
Do we still want this in 8.x? If so can we get a backport?

**Pull request merged**